share/p2p/peers/manager.go

```diff
@@ -75,7 +81,9 @@ func (s *Manager) Stop(ctx context.Context) error {
 	}
 }
 
-type DoneFunc func(success bool)
+type DoneFunc func(result sampleResult)
```

Comment:
does this have to be exposed?

Reply:
yes unless we make a shrexgetter package where both this and shrexgetter live